### ISSUE TITLE
Trouble getting format_on_save to work

### ISSUE THREAD

**Issue by Zolmok** [VonHeikemen/lsp-zero.nvim#198]
Here is my configuration.

```
local lsp = require("lsp-zero").preset({
	name = "recommended",
})

lsp.ensure_installed({
	"eslint",
	"rust_analyzer",
	"tsserver",
})

lsp.format_on_save({
	servers = {
		["eslint"] = { "javascript" },
                [ "lua_ls"] = {"lua"},
		["tsserver"] = { "typescript" },
		["rust_analyzer"] = { "rust" },
	},
})

lsp.setup()

local rust_lsp = lsp.build_options('rust_analyzer', {})

require('rust-tools').setup({server = rust_lsp})
```

When I write some sloppy looking code in either lua or rust and then save the file I thought `format_on_save` would run a formatter and fixup all of the code, but currently nothing is happening.  Is there something else I need to do or do I misunderstand how this is supposed to work?

**Comment by VonHeikemen** [comment]
That should work (works on my machine). The only thing missing is ignore the setup of rust_analyzer in lsp-zero. Add this before `lsp.setup()`

```lua
lsp.skip_server_setup({'rust_analyzer'})
```

You can use this command to verify the server can format the current buffer.

```lua
:lua require('lsp-zero.format').check('rust_analyzer')
```

**Comment by Zolmok** [comment]
It is working now. I tried it in a regular rust project and it works fine.  I was working on the rustlings project and for whatever reason it doesn't work there. I was having problems getting LSP to work there at all until I ran across these rust-analyzer settings: https://gist.github.com/jackos/7332fa8ab0b67a87f382fd566696f412 I think there is something strange with the rustlings project. TYVM for your help, I'm all set!
